src/updatechecker.cpp

```diff
@@ -219,9 +220,34 @@ UpdateChecker::UpdateChecker(): Thread("WinSparkle updates check")
 
 void UpdateChecker::Run()
 {
-    // no initialization to do, so signal readiness immediately
-    SignalReady();
+    while (1)
+    {
+        // no initialization to do, so signal readiness immediately
+        SignalReady();
+
+        bool checkUpdates;
+        if (Settings::ReadConfigValue("CheckForUpdates", checkUpdates))
+        {
+            if (checkUpdates)
+            {
+                time_t lastCheck = 0;
+                Settings::ReadConfigValue("LastCheckTime", lastCheck);
+                const time_t currentTime = time(NULL);
+
+                // Only check for updates in reasonable intervals:
+                const int interval = win_sparkle_get_update_check_interval();
+                if (currentTime - lastCheck >= interval)
+                {
+                    UpdateCheckWorker();
+                }
+            }
+        }
+        m_terminateEvent.WaitUntilSignaled(win_sparkle_get_update_check_interval());
+    }
+}
 
+void UpdateChecker::UpdateCheckWorker()
```

Comment:
This function isn’t really a “worker” (i.e. a thread that does some work), but a helper function to perform the task. `PerformUpdateCheck()` perhaps?